samza-yarn/src/main/java/org/apache/samza/job/yarn/LocalizerResourceMapper.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.samza.job.yarn;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.yarn.api.records.LocalResource;
+import org.apache.hadoop.yarn.api.records.LocalResourceType;
+import org.apache.hadoop.yarn.api.records.LocalResourceVisibility;
+import org.apache.hadoop.yarn.api.records.URL;
+import org.apache.hadoop.yarn.conf.YarnConfiguration;
+import org.apache.hadoop.yarn.util.ConverterUtils;
+import org.apache.hadoop.yarn.util.Records;
+import org.apache.samza.SamzaException;
+import org.apache.samza.config.Config;
+import org.apache.samza.config.YarnConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * A universal approach to generate local resource map which can be put in ContainerLaunchContext directly
+ */
+public class LocalizerResourceMapper {
+  private static final Logger log = LoggerFactory.getLogger(LocalizerResourceMapper.class);
+
+  private Config config; //job configurations
+  private YarnConfiguration yarnConfiguration; //yarn configurations
+  private Map<String, LocalResource> localResourceMap;
+
+  public LocalizerResourceMapper(Config config, YarnConfiguration yarnConfiguration) {
+    this.config = config;
+    this.yarnConfiguration = yarnConfiguration;
+    this.localResourceMap = new HashMap<>();
+  }
+
+  public void map() throws LocalizerResourceException {
```

Comment:
Don't need an explicit `map` here. I wonder if we can use invoke `map` internally inside the constructor. 

- Not sure I see a use-case to invoke `map` multiple times in this class.
- We can make `localResourceMap` immutable